include/drivers/timer/arm_generic.h

```diff
@@ -32,13 +32,22 @@ static inline void ackDeadlineIRQ(void)
 {
     ticks_t deadline = UINT64_MAX;
     setDeadline(deadline);
+    /* Ensure that the timer deasserts the IRQ before GIC EOIR/DIR.
```

Comment:
It's maybe a bit over the scope of this PR, but I wonder, why are we not simply disable the timer (`CNT_CTL.ENABLE`) here or mask the interrupt (`CNT_CTL.IMASK`)? That should also de-assert the interrupt (after an `isb()`). Is it a performance optimization that quasi-infinite value is used, or was it just a practical choice/hack? The git history is not clear, but interrupt masking usually seems the common way. 
 
The docs says the underlying counter keeps counting if `CNT_CTL.ENABLE` is cleared. Disabling it is explicitly mentioned to be a potential power saving option option. Also, we would not abuse `setDeadline()' then, it would only be used when somebody wants to actually set a deadline. It should have an `isb()` then also, we the high level calling contract seems to imply that once the function returns the deadline is active.

Reply:
The signal line might be floating if you disable the timer, you may also need to disable the timer interrupt in the GIC then. Masking would work, but may make verification harder because an additional hardware property needs to be modelled.

`setDeadline()` does not need an `isb()`, just writing the new value is enough to set a new deadline. Only here when the interrupt is supposed to be cleared is an `isb` necessary. The other callers don't have any race with the GIC.

You keep mentioning the calling contract, but with out-of-order execution and write buffering you're saying you want hard barriers at the end of every "contract" function.